### PR TITLE
31: Add Black Source Formatter

### LINKED ISSUE #31: Chore request: Add Black as source code formatter
Hey Joshua! 🌟

I've got a cool suggestion to make the coding life a bit easier. How about bringing [Black](https://github.com/psf/black) into this project?

### Why Black?

- **Consistent style**: It's like having an auto-stylist for the code, making everything look clean and uniform.
- **Focus on what matters**: Less time on formatting, more on coding the cool stuff.
- **Speedy reviews**: Code reviews become a breeze when we're not nitpicking on style.
- **Easy-peasy setup**: Integrates smoothly with tools we already use.

### Plan of action

1. Add Black to the toolbelt.
2. Give the code a one-time style makeover.

Would love to hear your thoughts on this!

### PR DESCRIPTION
**Pull request by jonathanlinat**
Resolves #31 

---

This PR aims to:

- add [Black](https://github.com/psf/black) as a new development dependency to format the project's source code and ensure consistency across the PY files
- format the project's source code
- add additional `makefile` commands to:
  * install the dependencies
  * format the source code
- update the `README` file to:
  * add additional instructions on how to use this project
  * adjust the details about the tools
- update the `.gitignore` file to ignore the packaged binaries (`.zip`)

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -1,23 +1,30 @@
 .PHONY: install uninstall reinstall test clean bsp2svg bsp2wad image2spr pak qmount spr2image unpak unwad wad
 
+prepare:
+	pip3 install -r requirements-dev.txt
+	pip3 install -r requirements.txt
+
+format:
+	python3 -m black ./
```
Comment (joshuaskelly) [inline makefile:8]: Python 2.7 was deprecated 3 years ago. Do we still need pip3/python3 on Linux machines? On windows both aliases point to the same thing.

Comment (jonathanlinat) [reply makefile:8]: Yeah, I'm not happy about that change, to be honest.

The `python` command still seems reserved for version 2.7, at least on Linux. I had to use `python3` instead, which is the default installed version on modern distros.

image

Comment (joshuaskelly) [reply makefile:8]: Okay. It is what is it. 🤷

**Comment by jonathanlinat** [comment]
@joshuaskelly What's missing to get this PR merged? Is there anything else I need to adjust?

**Pull request closed without merging**